typescript/sdk/src/token/Token.ts

```diff
@@ -232,24 +214,46 @@ export class Token implements IToken {
         token: addressOrDenom,
       });
     } else if (standard === TokenStandard.SealevelHypNative) {
+      assert(mailbox, `Mailbox required for Sealevel hyp tokens`);
       return new SealevelHypNativeAdapter(
         chainName,
         multiProvider,
-        sealevelAddresses!,
+        {
+          warpRouter: addressOrDenom,
+          mailbox,
+        },
         false,
       );
     } else if (standard === TokenStandard.SealevelHypCollateral) {
+      assert(mailbox, `Mailbox required for Sealevel hyp tokens`);
+      assert(
+        collateralAddressOrDenom,
+        `collateralAddressOrDenom required for Sealevel hyp collateral tokens`,
+      );
       return new SealevelHypCollateralAdapter(
         chainName,
         multiProvider,
-        sealevelAddresses!,
+        {
+          warpRouter: addressOrDenom,
+          token: collateralAddressOrDenom,
+          mailbox,
+        },
         false,
       );
     } else if (standard === TokenStandard.SealevelHypSynthetic) {
+      assert(mailbox, `Mailbox required for Sealevel hyp tokens`);
```

Comment:
nit: this `assert` is shared by all sealevel tokens so it could stay under `if (protocol === ProtocolType.Sealevel) {`

Reply:
I considered that but then TS won't know that we've confirmed the mailbox var isn't null. We'd still need the `!` symbols like before. Duplication vs type safety tradeoff. Opted for this approach but I can revert.